test/test_unary_ufuncs.py

```diff
@@ -377,6 +378,36 @@ def test_batch_vs_slicing(self, device, dtype, op):
 
         self.assertEqual(actual, expected)
 
+    @dtypes(*(torch.testing.get_all_int_dtypes() + torch.testing.get_all_fp_dtypes(include_bfloat16=False)))
+    def test_nan_to_num(self, device, dtype):
+        for contiguous in [False, True]:
+            for x in generate_numeric_tensors(dtype=dtype, device=device):
```

Comment:
I wonder if this test should use a different generator with more non-finite values, since the numeric tensors will be tested elsewhere?

Reply:
I think this should be fine as the generator has tensors with extremals, otherwise we'll will be creating almost similar function which does the same thing (takes dtype and device) and generates tensor. Let me know what you feel about that.

Reply:
I think you can just call make_tensor:

https://github.com/pytorch/pytorch/blob/7cde662f08b226fb8d7f8e99dbd6f40dbe2dca16/torch/testing/_internal/common_utils.py#L1432

And add the extremal values if it's float. You could create a smaller 64x64 tensor, for example. generate_numeric_tensors will create a very large tensor.

Reply:
Oh. Sure. Thanks!